### PR TITLE
Escape >, <, and & in PR title

### LINKED ISSUE #10: Escape Slack messages
A recent PR i opened was posted to slack with some bad formatting:
image

Original PR link: https://github.com/iFixit/server-templates/pull/3090

It looks like the `>` character in the PR title broke the slack message formatting.

We should try to do some escaping, or maybe strip out certain special characters, from the github input?

CC @addison-grant

### PR DESCRIPTION
**Pull request by addison-grant**
This is escaping necessity is explained here:
https://api.slack.com/reference/surfaces/formatting#escaping

I tested it under the 'mrkdwn' section here:
https://app.slack.com/block-kit-builder/T025FUEFN

Closes https://github.com/iFixit/PullBot/issues/10

### CODE REVIEW

File: pullbot.rb [hunk 1]
```diff
@@ -58,7 +58,8 @@ def make_json(repo_meta, pr, author_meta, action, sender_meta, pr_icon_uri, atta
                          "text" => {
                               "type" => "mrkdwn",
                               "text" => "*[<#{repo_meta['html_url']}|#{repo_meta['full_name']}>]*"\
-                                        " <#{pr['html_url']}|_#{pr['title']}_ *##{pr['number']}*>"
+                                        " <#{pr['html_url']}|_#{pr['title'].gsub(/&/,'&amp;').gsub(/>/,'&gt;').gsub(/</,'&lt;')}"\
+                                        "_ *##{pr['number']}*>"
```
Comment (addison-grant) [inline pullbot.rb]:
```suggestion
                                        " <#{pr['html_url']}|_#{pr['title'].gsub(/&/,'&amp;').gsub(/>/,'&gt;').gsub(/</,'&lt;')}_ "\
                                        "*##{pr['number']}*>"
```

**Comment by addison-grant** [comment]
A little easier to read the change with https://github.com/iFixit/PullBot/pull/11/files?diff=unified

**Review by djmetzle — APPROVED** [review]
CR :gear:

**Comment by addison-grant** [comment]
QA
===

Pullbot runs out of cominor at `/opt/pullbot/`.
You can 
1. clone this repo, and copy the config over from `/opt/pullbot/config.json` into your clone.
2. from your clone root run the `exec/build.sh` script 
3. from your clone root run the `exec/restart-with-latest.sh` script

Check messages in the Slack channel #pull-requests

Suggestion:
- open a test PR with the titles from the issue.

**Comment by GrantMatejka** [comment]
CR 🌽

**Comment by ardelato** [comment]
dev_block ☁️ 

It seems that `<` and `>` break the hyperlink formatting in the Slack message

| With single `>`	|
|:---:|
|Arrow characters break link	|


| With single `<`	|
|:---:|
|escaping character|

| With single `&`	|
|:---:|
|Ampersand escaping|


## Steps Taken:
1. In my Cominor home directory I cloned this repo, and copied the config over from `/opt/pullbot/config.json` into the cloned repo
2. Then ran the `exec/build.sh` script in the cloned repo
3. Then ran the `exec/restart-with-latest.sh` script
4. I then checked out the `test-branch` already in this repo
5. I created a new file and followed the process to pushing the change
6. I then created a new pull request with the title containing the escape characters https://github.com/iFixit/PullBot/pull/12
7. When I created the Pull Request I confirmed that the PR title showed up in the `#pull-requests` slack channel 
8. I noticed that the hyperlink only covered the first character in the title so tested the escaping characters independently and found that `<` and `>` break the hyperlink syntax
   - I did this by editing the tile before `closing` or `reopening` the same pull request
9. I also tested that a regular title still works without issues 👍 
Regular title link works

**Comment by ardelato** [comment]
QA 👍 

- [x] Titles with `<` , `>` , and/or `&` characters no longer break the slack syntax

## Steps Taken:
1. In my Cominor home directory I cloned this repo, and copied the config over from `/opt/pullbot/config.json` into the cloned repo
2. I then checked out this branch in the repo
2. Then ran the `exec/build.sh` script
3. Then ran the `exec/restart-with-latest.sh` script
4. I then checked out the `test-branch` already in this repo
5. I created a new file and followed the process to pushing the change
6. I then created a new pull request with the title containing the escape characters https://github.com/iFixit/PullBot/pull/12
7. When I created the Pull Request I confirmed that the PR title showed up in the `#pull-requests` slack channel 
8. I confirmed each character was escaping properly
   - I did this by editing the title before `closing` or `reopening` the same pull request
Works

9. I also tested that a regular title still works without issues 👍 
Regular title link works

**Pull request merged**